src/client/BaseApi.ts

```diff
@@ -141,16 +141,39 @@ export class BaseApi implements IAstarApi {
         return this._api.createType('Call', callHex);
     }
 
-    protected async connect() {
-        // establish node connection with the endpoint
-        if (!this._api) {
-            const provider = new WsProvider(this.endpoint);
-            const api = new ApiPromise({ provider });
-            const apiInst = await api.isReady;
-            this._api = apiInst;
+    protected async connect(networkIndex?: number): Promise<ApiPromise> {
```

Comment:
I noticed that (similar to before the change), every time the class runs a method, this function is called too. How about naming it `ensureConnection` or something like that.

Reply:
Fixed in [this](https://github.com/AstarNetwork/astar-token-api/pull/36/commits/bb8ced8f773904c87a2e0b6a424c531ba7caa694) commit